rpcs3/rpcs3qt/settings_dialog.cpp

```diff
@@ -1481,6 +1481,16 @@ settings_dialog::settings_dialog(std::shared_ptr<gui_settings> gui_settings, std
 		const auto [text, value] = get_data(ui->netStatusBox, index);
 		ui->gb_edit_dns->setEnabled(static_cast<np_internet_status>(value) != np_internet_status::disabled);
 		ui->enable_upnp->setEnabled(static_cast<np_internet_status>(value) != np_internet_status::disabled);
+
+		if (static_cast<np_internet_status>(value) == np_internet_status::disabled)
+		{
+			ui->psnStatusBox->setCurrentIndex(find_item(ui->psnStatusBox, static_cast<int>(g_cfg.net.psn_status.def)));
```

Comment:
I think this should not be changed here but instead the config value should be adjusted in system.cpp.
There's some function used to fixup config values